File: apps/cargo-tracker/app/cargotracker/estimation/application/queryservices/EstimateQueryService.scala
package cargotracker.estimation.application.queryservices

import cargotracker.estimation.domain.model.aggregates.Estimate
import cargotracker.estimation.domain.model.repositories.EstimateRepository
import cargotracker.estimation.domain.model.valueobjects.EstimateId

import javax.inject.{Inject, Singleton}

/** 輸送見積の参照系（CQRS Query 側）。 */
@Singleton
class EstimateQueryService @Inject() (repository: EstimateRepository):

  def findAll(): Seq[Estimate] = repository.findAll()

  def findById(estimateId: String): Option[Estimate] =
    repository.findById(EstimateId.unsafeFrom(estimateId))

File: apps/cargo-tracker/src/main/java/com/example/cargotracker/estimation/application/internal/queryservices/EstimateQueryService.java
package com.example.cargotracker.estimation.application.internal.queryservices;

import java.util.List;

/** 見積の読み取り（CQRS のクエリ側）。 */
public interface EstimateQueryService {

    /**
     * 見積の一覧（新しい順）。
     *
     * <p><strong>期限切れの判定は読み出しのたびに行う</strong>
     * （`domain-model.md` のビジネスルール 7。ADR-019 と同じ形）。
     */
    List<EstimateSummaryView> findAll();
}
